FreeRTOS/FreeRTOS
# Add feature to set credentials using buffer in MQTT_Mutual_Auth_wolfSSL demo

Description
-----------
In the existing MQTT_Mutual_Auth_wolfSSL demo, client credentials are loaded only from the file system. To enhance usability, this PR adds the ability to use buffers to load credentials. 

Test Steps
-----------

If the user wants to use a buffer to load device credentials such as certificates and private keys,
1.  Prepare a source file containing the contents of the credentials
2.  Add the file to the demo project.
3.  Define the newly introduced democonfigCREDENTIALS_IN_BUFFER macro in demo_config.h
4. Associate the variable name which holds credential contents with the following credential macro.

  - democonfigROOT_CA_PEM
  - democonfigCLIENT_CERTIFICATE_PEM
  - democonfigCLIENT_PRIVATE_KEY_PEM

5. Build and run the demo. 
Detailed instructions are included in the README_wolfSSL.md. 

Related Issue
-----------



By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.
